dagsync/subscriber.go

```diff
@@ -406,7 +407,7 @@ func (s *Subscriber) Sync(ctx context.Context, peerID peer.ID, nextCid cid.Cid,
 
 	var peerAddrs []multiaddr.Multiaddr
 	if peerAddr != nil {
-		peerAddrs = []multiaddr.Multiaddr{peerAddr}
+		peerAddrs = []multiaddr.Multiaddr{rmPeerIDFromMultiaddr(peerAddr)}
```

Comment:
The code would be much less verbose if we use `peer.SplitAddr`; e.g.:

```suggestion
		transport, _ := peer.SplitAddr(peerAddr)
		peerAddrs = []multiaddr.Multiaddr{rmPeerIDFromMultiaddr(transport)}
```

Reply:
Agree, will update.